keyboards/coban/pad3a/keymaps/via/keymap.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright 2021 RyanDam (https://github.com/RyanDam)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(KC_MUTE, KC_MNXT, KC_MPLY, KC_VOLD, KC_VOLU),
+};
+
+#if defined(VIA_ENABLE) && defined(ENCODER_ENABLE)
+
+#define ENCODERS 1
+static uint8_t  encoder_state[ENCODERS] = { 0 };
+static keypos_t encoder_cw[ENCODERS]    = {{ 3, 0 }};
+static keypos_t encoder_ccw[ENCODERS]  = {{ 4, 0 }};
+
+void encoder_action_register(uint8_t index, bool clockwise) {
+    keyevent_t encoder_event = (keyevent_t) {
+        .key = clockwise ? encoder_cw[index] : encoder_ccw[index],
+        .pressed = true,
+        .time = (timer_read() | 1)
+    };
+    encoder_state[index] = (clockwise ^ 1) | (clockwise << 1);
+    action_exec(encoder_event);
+}
+
+void encoder_action_unregister(void) {
+    for (int index = 0; index < ENCODERS; ++index) {
+        if (encoder_state[index]) {
+            keyevent_t encoder_event = (keyevent_t) {
+                .key = encoder_state[index] >> 1 ? encoder_cw[index] : encoder_ccw[index],
+                .pressed = false,
+                .time = (timer_read() | 1)
+            };
+            encoder_state[index] = 0;
+            action_exec(encoder_event);
+        }
+    }
+}
+
+void matrix_scan_user(void) {
+    encoder_action_unregister();
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    encoder_action_register(index, clockwise);
+    return false;
+};
+
+#endif
```

Comment:
[Encoder Map](https://docs.qmk.fm/#/feature_encoders?id=encoder-map) should be used instead. This also means removing all the other unnecessary stuff, like correcting the layout macro. 
```suggestion
```

Reply:
Thank you for supporting me on this PR.
This keyboard allows user to remap encoder turn behaviour via VIA, and this code allow it by overriding encoder callback event to propriate key in keymaps layout (key index 3 and 4). 
I'm new to QMK and [Encoder Map](https://docs.qmk.fm/#/feature_encoders?id=encoder-map) seem not to allow me to implement what I want as Encoder Map is hardcoded and cannot be change via VIA.
Or am I missing something?

Reply:
VIA supports changing encoder (map) functionality on the fly. See ["docs"](https://github.com/qmk/qmk_firmware/pull/17734).